bezier-rs/lib/src/subpath.rs

```diff
@@ -0,0 +1,311 @@
+			curve_stroke_color: String::from("black"),
+			curve_stroke_width: 2.,
+			anchor_stroke_color: String::from("black"),
+			anchor_stroke_width: 2.,
+			anchor_radius: 4.,
+			anchor_fill: String::from("white"),
+			handle_line_stroke_color: String::from("grey"),
+			handle_line_stroke_width: 1.,
+			handle_point_stroke: String::from("grey"),
+			handle_point_stroke_width: 1.5,
+			handle_point_radius: 3.,
+			handle_point_fill: String::from("white"),
+		}
+	}
+}
+
+impl SubPath {
+	/// Create a new SubPath using a list of ManipulatorGroups.
+	/// A SubPath with less than 2 ManipulatorGroups may not be closed.
+	pub fn new(manipulator_groups: Vec<ManipulatorGroup>, closed: bool) -> SubPath {
+		assert!(!closed || manipulator_groups.len() > 1);
+		SubPath { manipulator_groups, closed }
+	}
+
+	/// Create a subpath consisting of 2 manipulator groups from a bezier.
+	pub fn from_bezier(bezier: Bezier) -> Self {
+		SubPath::new(
+			vec![
+				ManipulatorGroup {
+					anchor: bezier.start(),
+					in_handle: None,
+					out_handle: bezier.handle_start(),
+				},
+				ManipulatorGroup {
+					anchor: bezier.end(),
+					in_handle: bezier.handle_end(),
+					out_handle: None,
+				},
+			],
+			false,
+		)
+	}
+
+	/// Returns true if and only if the subpath contains at least one manipulator point
+	pub fn is_empty(&self) -> bool {
+		self.manipulator_groups.is_empty()
+	}
+
+	/// Returns the number of ManipulatorGroups contained within the subpath.
+	pub fn len(&self) -> usize {
+		self.manipulator_groups.len()
+	}
+
+	/// Returns an iterator of the Beziers along the subpath.
+	pub fn iter(&self) -> SubPathIter {
+		SubPathIter { sub_path: self, index: 0 }
+	}
+
+	pub fn to_svg(&self, options: ToSVGOptions) -> String {
+		if self.is_empty() {
+			return String::new();
+		}
+
+		let subpath_options = format!(r#"stroke="{}" stroke-width="{}" fill="transparent""#, options.curve_stroke_color, options.curve_stroke_width);
+		let anchor_options = format!(
+			r#"r="{}", stroke="{}" stroke-width="{}" fill="{}""#,
+			options.anchor_radius, options.anchor_stroke_color, options.anchor_stroke_width, options.anchor_fill
+		);
+		let handle_point_options = format!(
+			r#"r="{}", stroke="{}" stroke-width="{}" fill="{}""#,
+			options.handle_point_radius, options.handle_point_stroke, options.handle_point_stroke_width, options.handle_point_fill
+		);
+		let handle_line_options = format!(
+			r#"stroke="{}" stroke-width="{}" fill="transparent""#,
+			options.handle_line_stroke_color, options.handle_line_stroke_width
+		);
+
+		let anchor_circles = self
+			.manipulator_groups
+			.iter()
+			.map(|point| format!(r#"<circle cx="{}" cy="{}" {}/>"#, point.anchor.x, point.anchor.y, anchor_options))
+			.collect::<Vec<String>>();
+		let mut path_pieces = vec![format!("M {} {}", self[0].anchor.x, self[0].anchor.y)];
+		let mut handle_pieces = Vec::new();
+		let mut handle_circles = Vec::new();
+
+		for start_index in 0..self.len() + (self.closed as usize) - 1 {
```

Comment:
Based on our discussion: Refactor `Bezier` to have use a helper that returns the contents of the `<path>` of the svg. This will allow the `SubPath`'s `to_svg` to make use of an iterator and still have the entire `SubPath` in one `<path>` tag.